authorization/src/main/kotlin/com/ritense/authorization/AuthorizationSupportedHelper.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.authorization
+
+import org.springframework.context.ApplicationContext
+import org.springframework.context.ApplicationContextAware
+import org.springframework.core.ResolvableType
+
+object AuthorizationSupportedHelper: ApplicationContextAware{
+    private lateinit var applicationContext: ApplicationContext
+
+    override fun setApplicationContext(applicationContext: ApplicationContext) {
+        this.applicationContext = applicationContext
+    }
+
+    fun checkSupported(clazz: Class<*>) {
+        val beanNames = applicationContext.getBeanNamesForType(ResolvableType.forClassWithGenerics(AuthorizationSpecificationFactory::class.java, clazz))
+        if (beanNames.isEmpty()) {
+            throw AuthorizedResourceNotSupportedException(clazz)
```

Comment:
Shouldn't this be AuthorizationResourceNotSupported?

Reply:
Sure, seems better